verl/trainer/ppo/core_algos.py

```diff
@@ -106,6 +107,54 @@ def compute_gae_advantage_return(token_level_rewards: torch.Tensor, values: torc
     return advantages, returns
 
 
+# NOTE(sgm): this implementation only consider outcome supervision, where the reward is a scalar.
+def compute_grpo_outcome_advantage(token_level_rewards: torch.Tensor,
```

Comment:
We should give each prompt a uuid and use the uuid for grouping

Reply:
Yes, index is the uuid. I pass the uuid through the `index` param.